Packet++/src/PacketUtils.cpp

```diff
@@ -49,14 +55,16 @@ uint16_t computeChecksum(ScalarBuffer<uint16_t> vec[], size_t vecSize)
 	{
 		sum = (sum & 0xffff) + (sum >> 16);
 	}
-
 	PCPP_LOG_DEBUG("Sum before invert = " << sum << ", 0x" << std::uppercase << std::hex << sum);
 
-	sum = ~sum;
+	// To obtain the checksum we take the ones' complement of this result
+	uint16_t result = sum;
+	result = ~result;

```

Comment:
I'm not sure why the extra `result` variable is needed. What's wrong with the previous code?

Reply:
to have a proper print std::hex << result;

Reply:
oh ok, that makes sense 👍